pkg/config/remote/uptane/client.go

```diff
@@ -146,6 +162,107 @@ func (c *Client) update(response *pbgo.LatestConfigsResponse) error {
 	return c.verify()
 }
 
+func (c *CoreAgentClient) updateRepos(response *pbgo.LatestConfigsResponse) error {
+	err := c.targetStore.storeTargetFiles(response.TargetFiles)
+	if err != nil {
+		return err
+	}
+	c.directorRemoteStore.update(response)
+	c.configRemoteStore.update(response)
+	_, err = c.directorTUFClient.Update()
+	if err != nil {
+		return errors.Wrap(err, "failed updating director repository")
+	}
+	_, err = c.configTUFClient.Update()
+	if err != nil {
+		e := fmt.Sprintf("could not update config repository [%s]", configMetasUpdateSummary(response.ConfigMetas))
+		return errors.Wrap(err, e)
+	}
+	return nil
+}
+
+// NewHTTPClient creates a new uptane client that will fetch the latest configs from the server over HTTP(s)
+func NewHTTPClient(cacheDB *bbolt.DB, host, site, apiKey string, orgUUIDProvider OrgUUIDProvider, options ...ClientOption) (c *CDNClient, err error) {
+
+	transactionalStore := newTransactionalStore(cacheDB)
+	targetStore := newTargetStore(transactionalStore)
+	orgStore := newOrgStore(transactionalStore)
+
+	httpClient := &http.Client{}
+
+	c = &CDNClient{
+		host:                host,
+		configRemoteStore:   newCDNRemoteConfigStore(httpClient, host, site, apiKey),
+		directorRemoteStore: newCDNRemoteDirectorStore(httpClient, host, site, apiKey),
+		Client: &Client{
+			site:               site,
+			targetStore:        targetStore,
+			orgStore:           orgStore,
+			transactionalStore: transactionalStore,
+			orgUUIDProvider:    orgUUIDProvider,
+		},
+	}
+	for _, o := range options {
+		o(c.Client)
+	}
+
+	if c.configLocalStore, err = newLocalStoreConfig(transactionalStore, c.site, c.configRootOverride); err != nil {
+		return nil, err
+	}
+
+	if c.directorLocalStore, err = newLocalStoreDirector(transactionalStore, c.site, c.directorRootOverride); err != nil {
+		return nil, err
+	}
+
+	c.configTUFClient = client.NewClient(c.configLocalStore, c.configRemoteStore)
+	c.directorTUFClient = client.NewClient(c.directorLocalStore, c.directorRemoteStore)
+	return c, nil
+}
+
+// Update updates the uptane client and rollbacks in case of error
+func (c *CDNClient) Update() error {
+	c.Lock()
+	defer c.Unlock()
+	c.cachedVerify = false
+
+	// in case the commit is successful it is a no-op.
+	// the defer is present to be sure a transaction is never left behind.
+	defer c.transactionalStore.rollback()
+
+	err := c.update()
+	if err != nil {
+		c.configTUFClient = client.NewClient(c.configLocalStore, c.configRemoteStore)
+		c.directorTUFClient = client.NewClient(c.directorLocalStore, c.directorRemoteStore)
+		return err
+	}
+	return c.transactionalStore.commit()
+}
+
+// update updates the uptane client
```

Comment:
Thought: The fact that this is effectively identical makes me wonder if we can slightly restructure the abstraction so that we don't have to have two distinct types. The uptane check process should be identical in both cases. The only thing different is the remote store / local store mechanisms. I wonder if we can take how we populate the remote store for the RC backend and pull it out of our current abstraction so we don't have to make two types just to handle that specific case.